src/Security/Authentication/Negotiate/src/Internal/LdapAdapter.cs

```diff
@@ -116,10 +141,58 @@ private static void GetNestedGroups(LdapConnection connection, ClaimsIdentity pr
             var group = searchResponse.Entries[0]; // Get the object that was found on ldap
             var groupDN = group.DistinguishedName;
 
+            if (processedGroups.Contains(groupDN)) {
+                // No need to continue, this group was resolved before
+                return;
+            }
+
+            retrievedClaims.Add(new KeyValuePair<string, string>(principal.RoleClaimType, groupCN));
             processedGroups.Add(groupDN);
 
+            // Get the group SID
+            var groupSID = group.Attributes["objectsid"];
+            if (groupSID is not null)
+            {
+                if (groupSID.Count == 1)
+                {
+                    // For some reason it is sometimes string and sometimes byte[] when it is returned as a string, then every byte is converted to a char and simply put together as a string
```

Comment:
This seems like the wrong layer to fix this. Is this something we can fix in `LdapConnection` or `LdapPal` in the runtime repo?

@joperezr What do you think? I see you are the one that added Linux support with https://github.com/dotnet/runtime/pull/35380.